src/components/side_panel/find_and_replace/find_and_replace.ts

```diff
@@ -72,12 +81,58 @@ export class FindAndReplacePanel extends Component<Props, SpreadsheetChildEnv> {
 
     onMounted(() => this.focusInput());
 
+    useEffect(
+      () => {
+        this.focusButton(this.lastClickedButton);
+      },
+      () => [this.env.model.getters.getActiveSheetId()]
+    );
+
+    useEffect(
+      () => {
+        if (
+          this.state.searchOptions.searchScope === "specificRange" &&
+          !this.state.searchOptions.specificRange
+        ) {
+          this.state.searchOptions.specificRange = this.lastSpecificRange;
+        } else if (
+          this.state.searchOptions.searchScope !== "specificRange" &&
+          this.state.searchOptions.specificRange
+        ) {
+          this.lastSpecificRange = this.state.searchOptions.specificRange;
+          this.state.searchOptions.specificRange = undefined;
+        }
+        this.updateSearch();
```

Comment:
Couldn't you just conserve the value of specificRange at all times and not send it in the command payload when `searchScopre !== "specificRange"`?